### PR TITLE
Global groupGenerator method

### LINKED ISSUE #426: Add group generator to the context
Add GENERATOR constant which contains standardized generator of the secp256k1 group to the context

### PR DESCRIPTION
**Pull request by aslesarenko**
closes #426

### CODE REVIEW

File: src/main/scala/sigmastate/serialization/OpCodes.scala [hunk 1]
```diff
@@ -191,5 +186,11 @@ object OpCodes extends ValueCodes {
...
 
   val ContextCode             : OpCode = (LastConstantCode + 142).toByte
-  val XorOfCode               : OpCode = (LastConstantCode + 143).toByte
+  val XorOfCode               : OpCode = (LastConstantCode + 143).toByte // equals to 255
+
+  
+  /** This is the maximum valid opcode in the current version.
+    * It is used by soft-forkability checks.
+    */
+  val MaxOpCode             : OpCode = XorOfCode
```
Comment (catena2w) [inline src/main/scala/sigmastate/serialization/OpCodes.scala]: For now it's usage is unclear, as soon as we have gaps between opcodes and soft-forkability is not implemented yet

Comment (aslesarenko) [reply src/main/scala/sigmastate/serialization/OpCodes.scala]: The intention was to compress opcodes, but since we don't do this, this can be removed.

Comment (catena2w) [reply src/main/scala/sigmastate/serialization/OpCodes.scala]: remove please

Comment (aslesarenko) [reply src/main/scala/sigmastate/serialization/OpCodes.scala]: removed

File: src/main/scala/sigmastate/eval/CostingRules.scala [hunk 2]
```diff
@@ -41,6 +41,18 @@ trait CostingRules extends SigmaLibrary { IR: RuntimeCosting =>
...
+  /** Implements basic costing rule invocation mechanism.
+    * Each MethodCall node of ErgoTree is costed using the same mechanism.
+    * When MethodCall is matched during traverse of ErgoTree in `RuntimeCosting.evalNode`:
+    * 1) the type of the receiver object is used to lookup the corresponding CostingHandler
+    * 2) The apply method of CostingHandler is called to create the Coster
+    * 3) When Coster is created, the costing-rule-method is looked up using reflection and then invoked.
+    * 4) The result of costing-rule-method is returned as the result of MethodCall node costing.
+    *
+    * Instances of this class are typically singleton objects (see below).
+    * Iach
```
Comment (kushti) [inline src/main/scala/sigmastate/eval/CostingRules.scala]: Looks like the writing has been suddenly aborted.

**Pull request merged**